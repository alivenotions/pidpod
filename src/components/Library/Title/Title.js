import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Heading = styled.h1`
  font-family: Arial, Helvetica, sans-serif;
  background-color: #cdced1;
  font-size: 1em;
  min-height: 2em;
  max-width: 10%;
  padding-top: 1em;
  padding-left: 1.5em;
  margin-left: 17%;
  overflow: auto;
`
Heading.displayName = 'Heading'
class Title extends React.Component {
  render() {
    return <Heading> {this.props.title} </Heading>
  }
}

Title.propTypes = {
  title: PropTypes.string.isRequired,
}

export default Title
