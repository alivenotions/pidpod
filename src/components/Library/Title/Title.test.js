import React from 'react'
import Title from './Title'
import { shallow } from 'enzyme'
describe('<Title />', () => {
  it('should be defined', () => {
    expect(Title).toBeDefined()
  })

  it('renders a Heading element', () => {
    const wrapper = shallow(<Title title="Podcasts" />)
    wrapper
      .find('Heading')
      .children()
      .debug()
    expect(wrapper.find('Heading')).toBeDefined()
  })

  it('renders the styled Heading correctly', () => {
    const wrapper = shallow(<Title title="Podcasts" />)
    expect(wrapper.find('Heading')).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const wrapper = shallow(<Title title="My Podcasts" />)
    expect(wrapper).toMatchSnapshot()
  })
})
