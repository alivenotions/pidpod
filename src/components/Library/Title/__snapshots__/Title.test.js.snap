// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders correctly 1`] = `
<Heading>
   
  My Podcasts
   
</Heading>
`;

exports[`<Title /> renders the styled Heading correctly 1`] = `
<Heading>
   
  Podcasts
   
</Heading>
`;
